lib/webpacker/node_bundler.rb

```diff
@@ -0,0 +1,67 @@
+# Finds available global or local node package manager - yarn or npm
+
+require "webpacker/configuration"
+
+module Webpacker::NodeBundler
+  extend self
+
+  def name
+    @package ||= packages.find(-> { raise Errno::ENOENT }) { |package| available?(package) }
+
+  rescue Errno::ENOENT
+    $stderr.puts "Node package managers npm and yarn not found! Scanned following paths for executables - #{packages}"
+    $stderr.puts "Make sure either npm or yarn is installed globally, or locally within your app node_modules folder."
+    exit!
+  end
+
+  def command
+    commands[name]
+  end
+
+  def command_dev
+    commands[:dev][name]
+  end
+
+  private
+
+    def available?(package)
+      null_device = Gem.win_platform? ? "/nul 2>&1" : "/dev/null 2>&1"
+      system("#{package} --version > #{null_device}")
```

Comment:
@justin808 The package manager is checked in order - `yarn`, `npm` (global) and then inside `node_modules` (local) - don't think using either affects the code so the env or config change isn't necessary. @ljharb true, unless of course they are already using it. NPM comes bundled as default with node so, thats "surely" available on all systems that has node.